kpops/components/base_components/pipeline_component.py

```diff
@@ -276,3 +278,15 @@ async def clean(self, dry_run: bool) -> None:
         :param dry_run: Whether to do a dry run of the command
         """
         await self.destroy(dry_run)
+
+    @model_serializer(mode="wrap", when_used="always")
+    def serialize_model(
+        self,
+        default_serialize_handler: pydantic.SerializerFunctionWrapHandler,
+        info: pydantic.SerializationInfo,
+    ) -> dict[str, Any]:
+        result = default_serialize_handler(self)
+        # HACK: workaround for exclude_none, which is otherwise evaluated too early
+        if info.exclude_none:
+            return exclude_by_value(result, None)
+        return result
```

Comment:
Have you considered letting `PipelineComponent` also inherit from `SerializeAsOptionalModel`? Then you wouln't have to duplicate this method AFAICS.

Reply:
good point! applied